arch/xtensa/src/esp32/Kconfig

```diff
@@ -259,6 +263,8 @@ config ESP32_WIRELESS
 	select NET
 	select ARCH_PHY_INTERRUPT
 	select ESP32_RNG
+	select ESP32_RT_TIMER
+	select ESP32_TIMER0
```

Comment:
Will it always be hard-coded to Timer0? Also I think in this case we need forbid the user from selecting Timer0 as general purpose timer!

Reply:
Because WiFi should use 1 hardware timer, so I think using timer0 directly and tell users that timer0 is used by WiFi and they should use other timers, this is also a good way.

Reply:
I think we should hide the Timer 0 instance  in case the user selected the WiFi. (and disable it in case it was already enabled).

Reply:
I think we can add extra API to get timer0, then there will be 2 APIs like `esp32_tim_init` and `esp32_tim0_init`. When WiFi is enable, `esp32_tim_init` can't return timer0, and only `esp32_tim0_init` return timer0.